app/src/main/java/fraktal/io/ext/EventBus.kt

```diff
@@ -0,0 +1,19 @@
+package fraktal.io.ext
+
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.asSharedFlow
+
+/**
+ * Simple EventBus.
+ * EventBus is not an EventStore. It does not store/save events to the storage/DB.
+ *
+ * https://elizarov.medium.com/shared-flows-broadcast-channels-899b675e805c
+ */
+class EventBus<E> {
```

Comment:
We have one reason why i defined channel for event on my last PR. we can get a doubling of same event in corner cases.  such as screen restoration, after sleep or after rotate, re-entering the screen and more.
therefore, I would like to leave the channel for the use of events

Reply:
OK. Roman has two examples for implementing Event Bus: https://elizarov.medium.com/shared-flows-broadcast-channels-899b675e805c .

This one is using channels:

```
class SingleShotEventBus {
    private val _events = Channel<Event>()
    val events = _events.receiveAsFlow() // expose as flow

    suspend fun postEvent(event: Event) {
        _events.send(event) // suspends on buffer overflow
    }
}
```
It is more or lees what you had in the initial source code.

Reply:
Added new commit, to use Channels (not Shared Flow) in EventBus implementation.